kedro/extras/datasets/geojson/geojson_dataset.py

```diff
@@ -0,0 +1,163 @@
+# You may obtain a copy of the License at
+#
+# http://www.apache.org/licenses/LICENSE-2.0
+#
+# THE SOFTWARE IS PROVIDED "AS IS", WITHOUT WARRANTY OF ANY KIND,
+# EXPRESS OR IMPLIED, INCLUDING BUT NOT LIMITED TO THE WARRANTIES
+# OF MERCHANTABILITY, FITNESS FOR A PARTICULAR PURPOSE, AND
+# NONINFRINGEMENT. IN NO EVENT WILL THE LICENSOR OR OTHER CONTRIBUTORS
+# BE LIABLE FOR ANY CLAIM, DAMAGES, OR OTHER LIABILITY, WHETHER IN AN
+# ACTION OF CONTRACT, TORT OR OTHERWISE, ARISING FROM, OUT OF, OR IN
+# CONNECTION WITH THE SOFTWARE OR THE USE OR OTHER DEALINGS IN THE SOFTWARE.
+#
+# The QuantumBlack Visual Analytics Limited ("QuantumBlack") name and logo
+# (either separately or in combination, "QuantumBlack Trademarks") are
+# trademarks of QuantumBlack. The License does not grant you any right or
+# license to the QuantumBlack Trademarks. You may not use the QuantumBlack
+# Trademarks or any confusingly similar mark as a trademark for your product,
+#     or use the QuantumBlack Trademarks in any other manner that might cause
+# confusion in the marketplace, including but not limited to in advertising,
+# on websites, or on software.
+#
+# See the License for the specific language governing permissions and
+# limitations under the License.
+
+"""GeoJSONDataSet loads and saves data to a local geojson file. The
+underlying functionality is supported by geopandas, so it supports all
+allowed geopandas (pandas) options for loading and saving geosjon files.
+"""
+import copy
+from pathlib import PurePosixPath
+from typing import Any, Dict, Union
+
+import fsspec
+import geopandas as gpd
+
+from kedro.io.core import (
+    AbstractVersionedDataSet,
+    DataSetError,
+    Version,
+    get_filepath_str,
+    get_protocol_and_path,
+)
+
+
+class GeoJSONDataSet(AbstractVersionedDataSet):
+    """``GeoJSONDataSet`` loads and saves data to a geojson file using an underlying filesystem
+    (eg: local, S3, GCS).
+    The underlying functionality is supported by geopandas, so it supports all
+    allowed geopandas (pandas) options for loading and saving geosjon files.
+
+    Example:
+    ::
+
+        >>> import geopandas as gpd
+        >>> from shapely.geometry import Point
+        >>> from kedro.io.contrib import GeoJSONDataSet
+        >>> data = gpd.GeoDataFrame({'col1': [1, 2], 'col2': [4, 5],
+        >>>                      'col3': [5, 6]}, geometry=[Point(1,1), Point(2,4)])
+        >>> # data_set = GeoJSONDataSet(filepath="gcs://bucket/test.geojson",
+        >>>                                save_args=None)
+        >>> data_set = GeoJSONDataSet(filepath="test.geojson",
+        >>>                                save_args=None)
+        >>> data_set.save(data)
+        >>> reloaded = data_set.load()
+        >>>
+        >>> assert data.equals(reloaded)
+
+    """
+
+    DEFAULT_LOAD_ARGS = {}  # type: Dict[str, Any]
+    DEFAULT_SAVE_ARGS = {"driver": "GeoJSON"}  # type: Dict[str, Any]
```

Comment:
`# type: Dict[str, Any]` is probably not necessary.

Reply:
Most other dataset seems to have this, unless I'm mistaken?

Reply:
When the dict is not empty ?

Reply:
But then the type inferred is `Dict[str, str]` by mypy, are we sure all the values are `str`s?